lib/matplotlib/pyplot.py

```diff
@@ -954,6 +954,10 @@ def subplot(*args, **kwargs):
         cbook._warn_external("The subplot index argument to subplot() appears "
                              "to be a boolean. Did you intend to use "
                              "subplots()?")
+    # Check for nrows and ncols, which are not valid subplot args:
+    if 'nrows' in kwargs or 'ncols' in kwargs:
+        raise TypeError("subplot does not accept nrows and/or ncols kwargs. "
```

Comment:
I think this should be
```suggestion
        raise TypeError("subplot() got an unexpected keyword argument 'nrows' and/or 'ncols'. "
```
Modulo wrapping. Compare

~~~
>>> def test():
...     pass

>>> test(ncols=1)
TypeError: test() got an unexpected keyword argument 'ncols'
~~~